ui/src/pages/gcp/login.js

```diff
@@ -303,62 +298,61 @@ class GCPLoginPage extends React.Component {
-                                roles contain your user.
-                                <br />
-                                E.g., if your account is 'jdoe', then{' '}
-                                <tt>gcp.fed.admin.user</tt> should have{' '}
-                                <tt>{USER_DOMAIN}.jdoe</tt> as a member.
-                            </p>
-                            <p>
-                                If that does not work, please ask your Athenz
-                                domain admin for assistance.
-                            </p>
-                        </ParentWrapperDiv>
-                    </div>
-                </CacheProvider>
-            );
-        }
-
-        return (
+        let gcpLoginContainer = displayProjects.length ? (
+            <div data-testid='gcp-login'>
+                <GcpHeader>
+                    <img src='/static/google-cloud.svg'></img>
+                </GcpHeader>
+                <ParentWrapperDiv>
+                    <form action='/gcp/login/post' method='post'>
+                        <input
+                            type='hidden'
+                            name='_csrf'
+                            value={this.props._csrf}
+                        ></input>
+                        <input
+                            type='hidden'
+                            name='isAdmin'
+                            value={this.props.isAdmin}
+                        ></input>
+                        <StyledP>Select a role:</StyledP>
+                        {displayProjects}
+                        <SubmitContainer>
+                            <Button type={'submit'}>Submit</Button>
+                        </SubmitContainer>
+                    </form>
+                </ParentWrapperDiv>
+            </div>
+        ) : (
+            <div data-testid='gcp-login-error'>
+                <GcpHeader>
+                    <img src='/static/google-cloud.svg'></img>
+                </GcpHeader>
+                <ParentWrapperDiv>
+                    <h3>
+                        Error: There are no GCP project roles associated with
+                        your account.
+                    </h3>
+                    <p>
+                        Check to make sure that your <tt>gcp.*</tt> roles
+                        contain your user.
+                        <br />
+                        E.g., if your account is 'jdoe', then{' '}
+                        <tt>gcp.fed.admin.user</tt> should have{' '}
+                        <tt>{USER_DOMAIN}.jdoe</tt> as a member.
+                    </p>
+                    <p>
+                        If that does not work, please ask your Athenz domain
+                        admin for assistance.
+                    </p>
+                </ParentWrapperDiv>
+            </div>
+        );
+        return this.props.isLoading.length !== 0 ? (
+            <ReduxPageLoader message={'Loading microsegmentation data'} />
```

Comment:
This error message doesnt look correct. Shouldnt it be something like "Loading GCP projects"?

Reply:
great catch, fixed.